likecoin/admin/matters.php

```diff
@@ -352,6 +352,64 @@ function likecoin_post_attachment_to_matters( $attachment_id ) {
 	return $matters_attachment_id;
 }
 
+/**
+ * Upload a file as draft image to matters
+ *
+ * @param int|     $image_url image url to be uploaded to matters.
+ * @param object | $image_infos all images' information of the post.
+ */
+function likecoin_post_url_image_to_matters( $image_url, $image_infos ) {
+	global $post;
+	$post_id              = $post->ID;
+	$file_path            = $image_url;
+	$file_path_components = preg_split( '#[.]#', $file_path );
+	$file_mime_type       = 'image/' . end( $file_path_components );
```

Comment:
We can use [get-headers](https://www.php.net/manual/en/function.get-headers.php) to reject non-image file and have more reliable result for use.